cmd/ipfs-cluster-service/main.go

```diff
@@ -284,23 +290,25 @@ multiaddresses.
 					)
 
 					// --force allows override of the prompt
-					if !c.GlobalBool("force") {
+					if !c.Bool("force") {
 						if !yesNoPrompt(confirm) {
 							return nil
 						}
 					}
 				}
 
+				// Set url. If exists, it will be the only thing saved.
+				cfgHelper.Manager().Source = c.Args().First()
```

Comment:
Note no other configuration is written (if source is present, only source is written).

It is true the Component Configurations objects are "defaulted". I wanted to change this into doing it only when no arguments are present. However, this is necessary for the `GetPeerstorePath()` below to work and save the peerstore.

This would also adds IFs blocks complicating the flow a bit so I don't think it's worth it.

I have made sure, however, that we do not ask for a custom secret if using a Source, even if the flag is provided.